tests-arquillian/src/test/java/org/jboss/weld/tests/alternatives/weld2313/VetoedAlternativeWithoutPropertyTest.java

```diff
@@ -0,0 +1,78 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.tests.alternatives.weld2313;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.AmbiguousResolutionException;
+import javax.enterprise.inject.UnsatisfiedResolutionException;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.util.TypeLiteral;
+import javax.inject.Inject;
+
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.impl.BeansXml;
+import org.jboss.weld.config.ConfigurationKey;
+import org.jboss.weld.config.WeldConfiguration;
+import org.jboss.weld.manager.BeanManagerImpl;
+import org.jboss.weld.test.util.Utils;
+import org.jboss.weld.tests.util.PropertiesBuilder;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Benjamin Confino
+ */
+@RunWith(Arquillian.class)
+public class VetoedAlternativeWithoutPropertyTest {
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    @Deployment(name = "VetoedAlternativeWithoutPropertyWar", managed = false)
+    public static WebArchive createTestArchive() {
+        BeansXml beansXml = new BeansXml().alternatives(MockPaymentProcessorImpl.class).stereotype(AlternativeConsumerStereotype.class);
+        WebArchive testDeployment = ShrinkWrap
+                .create(WebArchive.class, Utils.getDeploymentNameAsHash(VetoedAlternativeWithoutPropertyTest.class, Utils.ARCHIVE_TYPE.WAR))
+                .addPackage(VetoedAlternativeWithoutPropertyTest.class.getPackage())
+                .addAsServiceProvider(Extension.class, VetoingExtension.class)
+                .addAsWebInfResource(beansXml, "beans.xml");
+        return testDeployment;
+    }
+
+    @Test
+    public void vetoedAlternativesShouldBeRejected() {
+        try {
+            deployer.deploy("VetoedAlternativeWithoutPropertyWar");
+            Assert.fail("A web application deployed successfuly with a vetoed bean defined as an alternative in the beans.xml and the property org.jboss.weld.bootstrap.allowExcludedAlternatives was left as it's default value");
+        } catch (Exception e) {
+            Assert.assertTrue(e.getMessage().contains("WELD-001422"));
```

Comment:
This assertion probably fails because when running in-container Arquillian tests the `DeploymentExceptionTransformer` is used to extract the final exception and message (see also https://github.com/weld/core/blob/master/tests-arquillian/src/test/java/org/jboss/weld/tests/util/WildFly8DeploymentExceptionTransformer.java).

I would leave out this assertion completely. Also I would replace the `Deployer` API with `@org.jboss.arquillian.container.test.api.ShouldThrowException`.